### DISCUSSION TITLE
在Ant-Design-Pro V6中使用@umijs/max的request响应401时错误处理

### DISCUSSION THREAD

**Discussion by ycz-dev** [umijs/umi#12252] [Q&A]
在使用request时mock返回了401状态码，并没有经过拦截器和errorThrower直接进入errorHandler处理这是为什么

**Comment by fz6m** [comment]
非 200 应该会直接错误，这个是 use request 底层用的 axios 的默认性质决定的，默认情况下，axios 对非 200 状态码的响应都默认认为是错误的响应，会直接拋错，如果想避免这种行为，需要配置一个 `validateStatus: () => true` ，或者在这个函数里筛选下会报错的状态码（详见 axios 文档）。

所以现在基本上所有的请求都是 200 响应码，但是根据响应里的一个 `code` 字段来区分是否失败。
